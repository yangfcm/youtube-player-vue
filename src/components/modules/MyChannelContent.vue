<template>
  <div>
    <app-loader v-if="!channels && !error"></app-loader>
    <app-error-message v-if="error">{{ error }}</app-error-message>
    <div v-if="channels && !error" id="my-channel-content">
      <app-page-title>My Subscriptions</app-page-title>
      <app-channel-grid :channels="channels.items"></app-channel-grid>
      <app-blank></app-blank>
      <app-more-button
        :nextPageToken="channels.nextPageToken"
        :isLoadingMore="isLoadingMore"
        @onClickMore="handleMore($event)"
      >More channels</app-more-button>
    </div>
  </div>
</template>

<script>
import { mapState, mapActions, mapGetters } from "vuex";
import Loader from "../common/Loader";
import ErrorMessage from "../common/ErrorMessage";
import PageTitle from "../common/PageTitle";
import MoreButton from "../modules/MoreButton";
import ChannelGrid from "../modules/ChannelGrid";
import Blank from "../common/Blank";

export default {
  components: {
    appLoader: Loader,
    appErrorMessage: ErrorMessage,
    appPageTitle: PageTitle,
    appMoreButton: MoreButton,
    appChannelGrid: ChannelGrid,
    appBlank: Blank,
  },
  data: function () {
    return {
      channels: null,
      error: "",
      isLoadingMore: false,
    };
  },
  computed: {
    ...mapState(["channel"]),
    ...mapGetters(["channelErrorMessage"]),
  },
  methods: {
    ...mapActions(["fetchMyChannels"]),
    async handleMore($event) {
      if (this.isLoadingMore) return;
      this.isLoadingMore = true;
      const nextPageToken = $event;
      await this.fetchMyChannels(nextPageToken);
      if (this.channel.channelError) {
        this.error = this.channelErrorMessage;
      } else if (this.channel.myChannels) {
        this.channels = {
          ...this.channels,
          items: this.channels.items.concat(this.channel.myChannels.items),
          nextPageToken: this.channel.myChannels.nextPageToken,
        };
        this.error = "";
      }
      this.isLoadingMore = false;
    },
  },
  async created() {
    await this.fetchMyChannels();
    if (this.channel.channelError) {
      this.error = this.channelErrorMessage;
    } else if (this.channel.myChannels) {
      this.channels = this.channel.myChannels;
      this.error = "";
    }
  },
};
</script>

<style scoped>
</style>