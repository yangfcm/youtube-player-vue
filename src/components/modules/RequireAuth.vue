<template>
  <div class="ui red message">
    <app-blank></app-blank>
    <div class="ui header aligned center">You have not signed in</div>
    <app-blank></app-blank>
    <div class="ui header aligned center">
      <button class="ui youtube button negative basic" @click="handleSignIn">
        <i class="youtube icon"></i>
        Sign in
      </button>
    </div>
    <app-blank></app-blank>
  </div>
</template>

<script>
import googleAuth from "../../mixins/googleAuth";
import Blank from "../common/Blank";
export default {
  components: {
    appBlank: Blank,
  },
  mixins: [googleAuth],
  methods: {
    handleSignIn() {
      this.handleGoogleSignin();
    },
  },
};
</script>

<style scoped>
</style>