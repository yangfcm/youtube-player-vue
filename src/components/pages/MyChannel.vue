<template>
  <!-- MyChannel page - Channels subscribed by authed user -->
  <div>My channel</div>
</template>

<script>
export default {};
</script>

<style scoped></style>
