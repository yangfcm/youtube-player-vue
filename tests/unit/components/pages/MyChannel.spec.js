import { shallowMount } from "@vue/test-utils";
import MyChannel from "@/components/pages/MyChannel";
import RequireAuth from "@/components/modules/RequireAuth";
import MyChannelContent from "@/components/modules/MyChannelContent";

describe("Test MyChannel page", () => {
  test("should show RequireAuth component if user does not log in", () => {
    const $store = {
      getters: {
        signedIn: false,
      },
    };
    const wrapper = shallowMount(MyChannel, {
      mocks: {
        $store,
      },
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.findComponent(RequireAuth).exists()).toBe(true);
    expect(wrapper.findComponent(MyChannelContent).exists()).toBe(false);
  });

  test("should show MyChannelContent component if user logs in", () => {
    const $store = {
      getters: {
        signedIn: true,
      },
    };
    const wrapper = shallowMount(MyChannel, {
      mocks: {
        $store,
      },
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.findComponent(RequireAuth).exists()).toBe(false);
    expect(wrapper.findComponent(MyChannelContent).exists()).toBe(true);
  });
});
