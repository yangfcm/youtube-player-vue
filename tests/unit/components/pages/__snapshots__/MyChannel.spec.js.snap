// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MyChannel page should show MyChannelContent component if user logs in 1`] = `
<div>
  <app-user-banner-stub></app-user-banner-stub>
  <app-menu-stub></app-menu-stub>
  <app-my-channel-content-stub></app-my-channel-content-stub>
  <!---->
</div>
`;

exports[`Test MyChannel page should show RequireAuth component if user does not log in 1`] = `
<div>
  <app-user-banner-stub></app-user-banner-stub>
  <app-menu-stub></app-menu-stub>
  <!---->
  <app-require-auth-stub></app-require-auth-stub>
</div>
`;
